server/src/test/integration/org/apache/cassandra/sidecar/routes/NodeDecommissionIntegrationTest.java

```diff
@@ -0,0 +1,134 @@
+ */
+@ExtendWith(VertxExtension.class)
+public class NodeDecommissionIntegrationTest extends IntegrationTestBase
+{
+    @CassandraIntegrationTest(nodesPerDc = 2)
+    void decommissionNodeDefault(VertxTestContext context) throws InterruptedException
+    {
+        final String[] jobId = new String[1];
+        String testRoute = "/api/v1/cassandra/operations/decommission?force=true";
+        testWithClient(client -> client.put(server.actualPort(), "127.0.0.1", testRoute)
+                                       .send(context.succeeding(response -> {
+                                           logger.info("Response Status:" + response.statusCode());
+                                           OperationalJobResponse decommissionResponse = response.bodyAsJson(OperationalJobResponse.class);
+                                           assertThat(decommissionResponse.status()).isEqualTo(RUNNING);
+                                           jobId[0] = String.valueOf(decommissionResponse.jobId());
+                                       })));
+        Uninterruptibles.sleepUninterruptibly(10, TimeUnit.SECONDS);
+        pollStatusForState(jobId[0], SUCCEEDED, null);
+        context.completeNow();
+        context.awaitCompletion(2, TimeUnit.MINUTES);
+    }
+
+    @CassandraIntegrationTest(nodesPerDc = 2)
+    void decommissionNodeWithFailure(VertxTestContext context) throws InterruptedException
+    {
+        String testRoute = "/api/v1/cassandra/operations/decommission";
+        testWithClient(client -> client.put(server.actualPort(), "127.0.0.1", testRoute)
+                                       .send(context.succeeding(response -> {
+                                           logger.info("Response Status:" + response.statusCode());
+                                           logger.info("Response Body:" + response.bodyAsString());
+                                           OperationalJobResponse decommissionResponse = response.bodyAsJson(OperationalJobResponse.class);
+                                           assertThat(decommissionResponse.status()).isEqualTo(FAILED);
+                                           assertThat(decommissionResponse.jobId()).isNotNull();
+                                           String jobId = String.valueOf(decommissionResponse.jobId());
+                                           assertThat(jobId).isNotNull();
+                                           context.completeNow();
+                                       })));
+        context.awaitCompletion(2, TimeUnit.MINUTES);
+    }
+
+    private void pollStatusForState(String uuid,
+                                    OperationalJobStatus expectedStatus,
+                                    String expectedReason)
+    {
+        int attempts = 10;
+        String status = "/api/v1/cassandra/operational-jobs/" + uuid;
+        AtomicBoolean stateReached = new AtomicBoolean(false);
+        logger.info("Job Stats Attempt: {}", attempts);
```

Comment:
maybe you want to place it in the `loopAssert`? Right now, it only logs once with the `attempts` value 10